server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -1182,4 +1226,18 @@ private static String toUnderscoreCase(String value) {
         return sb.toString();
     }
 
+    private static class XContentRetainingException extends ElasticsearchException {
+
+        private final String fromXContentExceptionType;
```

Comment:
Would be nice to also return this type from the static `getExceptionName` method.